orangecontrib/educational/widgets/tests/test_owpolynomialclassification.py

```diff
@@ -446,104 +447,121 @@ def test_update_model(self):
             self.assertEqual(
-                len(self.get_output("Coefficients")), num_coefficients)
+                len(self.get_output(w.Outputs.coefficients)), num_coefficients)
 
         # manulay set LogisticRegression
-        self.send_signal("Learner", LogisticRegressionLearner())
+        self.send_signal(w.Inputs.learner, LogisticRegressionLearner())
 
         # to check correctness before degree is changed
         num_coefficients = sum(i + 1 for i in range(w.degree + 1))
-        self.assertEqual(len(self.get_output("Coefficients")), num_coefficients)
+        self.assertEqual(len(self.get_output(w.Outputs.coefficients)), num_coefficients)
 
         # change degree
         for j in range(1, 6):
             w.degree_spin.setValue(j)
             num_coefficients = sum(i + 1 for i in range(w.degree + 1))
             self.assertEqual(
-                len(self.get_output("Coefficients")), num_coefficients)
+                len(self.get_output(w.Outputs.coefficients)), num_coefficients)
 
     def test_send_data(self):
         """
         Check data output signal
         """
         w = self.widget
 
-        self.assertIsNone(self.get_output("Data"))
+        self.assertIsNone(self.get_output(w.Outputs.data))
 
-        self.send_signal("Data", self.iris)
+        self.send_signal(w.Inputs.data, self.iris)
 
         # check correct number of attributes
         for j in range(1, 6):
             w.degree_spin.setValue(j)
             self.assertEqual(
-                len(self.get_output("Data").domain.attributes), 2)
+                len(self.get_output(w.Outputs.data).domain.attributes), 2)
 
-        self.assertEqual(len(self.get_output("Data").domain.metas), 1)
-        self.assertIsNotNone(self.get_output("Data").domain.class_var)
+        self.assertEqual(len(self.get_output(w.Outputs.data).domain.metas), 1)
+        self.assertIsNotNone(self.get_output(w.Outputs.data).domain.class_var)
 
         # check again none
-        self.send_signal("Data", None)
-        self.assertIsNone(self.get_output("Data"))
+        self.send_signal(w.Inputs.data, None)
+        self.assertIsNone(self.get_output(w.Outputs.data))
 
     def test_send_report(self):
         """
         Just test everything not crashes
         """
         w = self.widget
 
-        self.send_signal("Data", self.iris)
+        self.send_signal(w.Inputs.data, self.iris)
         self.process_events(lambda: getattr(w, w.graph_name).svg())
         w.send_report()
+
+    def test_bad_learner(self):
+        """
+        Some learners on input might raise error.
+        GH-38
+        """
+        w = self.widget
+
+        self.assertFalse(w.Error.fitting_failed.is_shown())
+        learner = LogisticRegressionLearner()
+        learner.preprocessors = [Discretize()]
+        self.send_signal(w.Inputs.learner, learner)
+        self.send_signal(w.Inputs.data, self.iris)
+        self.assertTrue(w.Error.fitting_failed.is_shown())
```

Comment:
Why does this fail?

Reply:
@jerneju, @PrimozGodec Why does this fail?

Reply:
Classification requires at least two numerical features.